frontend/controller/actions/chatroom.js

```diff
@@ -247,6 +247,47 @@ export default (sbp('sbp/selectors/register', {
       ]
     })).then((keys) => [keys.filter(Boolean)])
   },
+  'gi.actions/chatroom/_ondeleted': async (contractID: string, state: Object) => {
+    const rootGetters = sbp('state/vuex/getters')
+    const identityState = rootGetters.currentIdentityState
+    if (identityState.chatRooms?.[contractID]) {
+      const identityContractID = rootGetters.ourIdentityContractId
+
+      await sbp('gi.actions/identity/deleteDirectMessage', { contractID: identityContractID, data: { contractID } }).catch(e => {
+        console.warn(`[handleDeletedContract] ${e.name} thrown by gi.actions/identity/deleteDirectMessage ${identityContractID} for ${contractID}:`, e)
+      })
+    } else {
+      // This is a group chatroom. To determine which group the chatroom
+      // belongs to, we need to go over each group, since there isn't a
+      // chatroom->group relationship stored.
+      const cIDs = Object.entries(identityState.groups || {}).filter(([cID, state]) => {
+        return !((state: any): Object).hasLeft
+      }).map(([cID]) => {
+        return cID
+      })
+
+      for (const cID of cIDs) {
+        const groupState = await sbp('chelonia/contract/state', cID).catch(() => {})
+        // If the chatroom isn't part of this group, continue
+        if (!groupState?.chatRooms?.[contractID]) continue
+
+        if (!groupState.chatRooms[contractID].deletedDate) {
+          // If the chatroom hasn't been 'deleted' in the group, attempt to do
+          // so now.
+          await sbp('gi.actions/group/deleteChatRoom', {
+            contractID: cID,
+            data: { chatRoomID: contractID }
+          }).catch(e => {
+            console.warn(`[handleDeletedContract] ${e.name} thrown by gi.actions/group/deleteChatRoom ${cID} for ${contractID}:`, e)
+          })
+        }
```

Comment:
One of the goals if you recall was to reuse the chatroom contract in other apps, including apps that might not have a group contract. Does this code break that requirement? If so, can it be written in a way that would still work with other apps that do not interact with the group contract?

Reply:
This isn't part of the contract so I don't see how the requirement is broken.